### PR TITLE
daemon: Use rpc for "version" command

### LINKED ISSUE #5994: monerod version command should get running daemon's version via RPC
One would expect[1] the command `monerod version` (as opposed to the _option_ `monerod --version`) to get the running daemon's version via RPC. Instead, it simply prints the version from the binary that was invoked, just like the `--version` option:

```
$ ./release-v0.14/release/bin/monerod version
2019-10-16 19:27:46.202     I Monero 'Boron Butterfly' (v0.14.1.2-release)
2019-10-16 19:27:46.208  I Generating SSL certificate
Monero 'Boron Butterfly' (v0.14.1.2-release)
$ ./master/release/bin/monerod version
2019-10-16 19:28:04.548   I Monero 'Boron Butterfly' (v0.14.1.2-d66db18c0)
2019-10-16 19:28:04.553        I Generating SSL certificate
Monero 'Boron Butterfly' (v0.14.1.2-d66db18c0)
```

No attempt at RPC is made when the command is `version` (this would fail with any other command, e.g. `status`):

```
$ ./master/release/bin/monerod --rpc-bind-port=0 version
2019-10-16 19:33:05.577   I Monero 'Boron Butterfly' (v0.14.1.2-d66db18c0)
2019-10-16 19:33:05.583        I Generating SSL certificate
Monero 'Boron Butterfly' (v0.14.1.2-d66db18c0)
```

It doesn't seem that an RPC to support this exists -- [get_version](https://www.getmonero.org/resources/developer-guides/daemon-rpc.html#get_version) doesn't return a version string, but rather some opaque unsigned int. It seems reasonable to add one, though. Of course, it should be disabled for restricted RPC. Any objections?

[1]: `monerod --help` says `Usage: ./master/release/bin/monerod [options|settings] [daemon_command...]`. So, when invoking `monerod version`, `version` is the `daemon_command` argument, not an "option."

### PR DESCRIPTION
**Pull request by ndorf**
Fixes #5994. To be useful, #6017 and #6015 are also needed.

### CODE REVIEW

File: src/daemon/rpc_command_executor.cpp [hunk 1]
```diff
@@ -2364,4 +2364,39 @@ bool t_rpc_command_executor::set_bootstrap_daemon(
...
+        }
+    }
+
+    if (res.version.empty())
+    {
+        tools::fail_msg_writer() << "The daemon software version is not available.";
+    }
+    else
+    {
+        tools::success_msg_writer() << res.version;
```
Comment (moneromooo-monero) [inline src/daemon/rpc_command_executor.cpp]: This lets a daemon controlled by a third party put something like this in the version string:
WARNING: your wallet is old and vulnerable to <insert bullshit here>. Please upgrade ASAP to https://pwn-me-now.com/fake-monero.tar.gz
Maybe some sanitization should be done on the version string first.

**Comment by ndorf** [comment]
Updated this to only print the version string if it matches a rather strict regex, and added a unit test to ensure that the current `MONERO_VERSION` passes, among other things.

I considered a more lenient check, but any remotely useful one would still need the unit test, at which point we might as well check very strictly.

File: src/daemon/rpc_command_executor.cpp (hunk 1 above)
Comment (ndorf) [reply src/daemon/rpc_command_executor.cpp]: How about now?

**Comment by ndorf** [comment]
Minor update; no reason to create an `std::string` for `fail_message`

**Comment by ndorf** [comment]
Another minor update; just making a temporary `regex` object is probably better than having a static one.

**Pull request merged**